FastSimulation/Calorimetry/src/CaloResponse.cc

```diff
@@ -0,0 +1,82 @@
+#include "FastSimulation/Calorimetry/interface/CaloResponse.h"
+
+CaloResponse::CaloResponse( const edm::ParameterSet& pset )
+{
+
+  // Get the filename and histogram name
+  std::string ecalScalesFileName_ = pset.getUntrackedParameter<std::string>("fileName");
+  std::string ecalScalesHistogramName_ = pset.getUntrackedParameter<std::string>("histogramName");
+
+  // Read histo
+  edm::FileInPath myDataFile( ecalScalesFileName_ ); // TODO: check if this is necessary
+  TFile * myFile = new TFile( myDataFile.fullPath().c_str(), "READ" );
+
+  gROOT->cd(); // create histogram in memory
+  auto obj = myFile->Get( ecalScalesHistogramName_.c_str() );
+
+  // Check if histogram exists in file
+  if(!obj) {
+    throw cms::Exception( "FileReadError" )
+      << "Histogram \"" << ecalScalesHistogramName_
+      << "\" not found in file \"" << ecalScalesFileName_
+      << "\", used for correcting the response of the ECAL in FastSim.\n";
+  }
+  h3_ = new TH3F( *((TH3F*)obj) );
+
+  myFile->Close();
+  delete myFile;
+
+}
+
+
+float CaloResponse::getScale( const RawParticle& particleAtEcalEntrance,
```

Comment:
I would remove this function.
To my taste it's too specific to the calorimetrymanager.
I would move all this code to the calorimetrymanager.